### ISSUE TITLE
[Bug]: Dimplex DL 50 WE2/WH2

### ISSUE THREAD

**Issue by tomtefuchs** [Jey-Cee/ioBroker.enocean#144]
### Contact Details

thomas@auchschoen.net

### What happened?

Trying to teach in the dimplex DL 50 WH2 via the pairing mechanism of iobroker.enocean. Device gets correctly recognized and EEP is displayed as D2-50-01. At the device, the pairing is NOT indicated via the y[channel] indicator and in the object browser of iobroker the device and channels are displayed, but no states/values are updated. 
Before I tried this way, I teached-in the device manually via the windows enocean client - that worked also for iobroker - states where updated, but commands where not possible - so I thought, a start from scratchwould be helpful ;)

Any ideas? Should the teach-in via iobroker.enocean should work with this device?

Thanks,
tomtefuchs

### Adapter Version

0.7.1

### JS-Controller Version

4.0.23

### Node JS Version

12.22.5

### NPM Version

8.16.0

### Operating System

Debian

**Comment by Jey-Cee** [comment]
> I teached-in the device manually via the windows enocean client

With the same EnOcean Stick? If yes did you reset the DL 50 before teachin in ioBroker? 
If not it was possible teached out instead teached in while this is the same procedure. 

Please look into: http://tools.enocean-alliance.org/EEPViewer/profiles/D2/50/01/D2-50-01.pdf there you will find the commands you can send and which parameters you have to set. All parameters must have a value that is not null. 
Parameters that doesn't have a object in ioBroker are set automatically by the adapter.

Also this device does only accept commands within 1 second after they have sent a telegram. This is also described in the PDF.

**Comment by tomtefuchs** [comment]
good idea. just resetted the device and removed the device from iobroker and then followed the procedure the adapter is providing. no success.

log file is showing this:

2022-08-28 18:41:58.614 - info: enocean.0 (386141) ["/dev/ttyACM0","/dev/ttyAMA0","/dev/ttyUSB0","/dev/serial/by-id/usb-EnOcean_GmbH_EnOcean_USB_300_DC_FT4T9VBN-if00-port0","/dev/serial/by-id/usb-dresden_elektronik_ingenieurtechnik_GmbH_ConBee_II_DE2478250-if00"]
2022-08-28 18:42:19.346 - debug: enocean.0 (386141) 55000d0701fdd480ff61000150d20586c9fd0000ffffffff4400
2022-08-28 18:42:19.386 - debug: enocean.0 (386141) teachinMethod: UTE from ID "0586c9fd"
2022-08-28 18:42:19.405 - info: enocean.0 (386141) Create objects for D2-50-01 from GLEN_DIMLAX_GMBH with id: 0586c9fd
2022-08-28 18:42:19.411 - debug: enocean.0 (386141) Sent data: 55000d0701fdd491ff61000150d2ffe85f8000010586c9fdff000d
2022-08-28 18:42:19.424 - debug: enocean.0 (386141) Sent data: 55000d0701fdd491ff61000150d2ffe85f8000010586c9fdff000d
2022-08-28 18:42:19.435 - debug: enocean.0 (386141) 55000100026500
2022-08-28 18:42:19.437 - debug: enocean.0 (386141) Packet type 2 received: {"code":"OK"}
2022-08-28 18:42:19.438 - debug: enocean.0 (386141) 55000100026500
2022-08-28 18:42:19.440 - debug: enocean.0 (386141) Packet type 2 received: {"code":"OK"}
2022-08-28 18:42:21.362 - debug: enocean.0 (386141) 55000d0701fdd480ff61000150d20586c9fd0000ffffffff4400
2022-08-28 18:42:21.373 - debug: enocean.0 (386141) Message for ID 0586c9fd has been received. It was repeated 0 times.
2022-08-28 18:42:23.407 - debug: enocean.0 (386141) 55000d0701fdd480ff61000150d20586c9fd0000ffffffff4000
2022-08-28 18:42:23.428 - debug: enocean.0 (386141) Message for ID 0586c9fd has been received. It was repeated 0 times.
2022-08-28 18:42:25.445 - debug: enocean.0 (386141) 55000d0701fdd480ff61000150d20586c9fd0000ffffffff4300
2022-08-28 18:42:25.497 - debug: enocean.0 (386141) Message for ID 0586c9fd has been received. It was repeated 0 times.
2022-08-28 18:42:27.484 - debug: enocean.0 (386141) 55000d0701fdd480ff61000150d20586c9fd0000ffffffff4300
2022-08-28 18:42:27.492 - debug: enocean.0 (386141) Message for ID 0586c9fd has been received. It was repeated 0 times.
2022-08-28 18:42:29.531 - debug: enocean.0 (386141) 55000d0701fdd480ff61000150d20586c9fd0000ffffffff4300
2022-08-28 18:42:29.544 - debug: enocean.0 (386141) Message for ID 0586c9fd has been received. It was repeated 0 times.
2022-08-28 18:42:31.561 - debug: enocean.0 (386141) 55000d0701fdd480ff61000150d20586c9fd0000ffffffff4300
2022-08-28 18:42:31.578 - debug: enocean.0 (386141) Message for ID 0586c9fd has been received. It was repeated 0 times.
2022-08-28 18:42:33.612 - debug: enocean.0 (386141) 55000d0701fdd480ff61000150d20586c9fd0000ffffffff4300
2022-08-28 18:42:33.654 - debug: enocean.0 (386141) Message for ID 0586c9fd has been received. It was repeated 0 times.
2022-08-28 18:42:35.654 - debug: enocean.0 (386141) 55000d0701fdd480ff61000150d20586c9fd0000ffffffff4300
2022-08-28 18:42:35.667 - debug: enocean.0 (386141) Message for ID 0586c9fd has been received. It was repeated 0 times.
2022-08-28 18:42:37.682 - debug: enocean.0 (386141) 55000d0701fdd480ff61000150d20586c9fd0000ffffffff4100
2022-08-28 18:42:37.687 - debug: enocean.0 (386141) Message for ID 0586c9fd has been received. It was repeated 0 times.
2022-08-28 18:42:44.496 - info: admin.0 (385990) <== Disconnect system.user.admin from ::ffff:192.168.178.22

**Comment by Jey-Cee** [comment]
Teachin isn't successfully. Looks like the DL50 does not recognize the answer from the adapter. 
Can't say whats the problem as telegrams are OK.

**Comment by tomtefuchs** [comment]
ok. thanks for checking the telegrams. I did the teach in again via the windows client (DOLPHIN VIEW). Here the speciality was, that I had to send a UTE Request with a genric destination id from the gateway first - than the teach in from the device came - and than I could respond to that - some how the device needs also the teach in request first ... weird.

"Last" question. I'm now checking again how to send commands to the device - and as You said, making sure that every required parameter is set. But how in the end can I send the command?

**Comment by Jey-Cee** [comment]
> But how in the end can I send the command?

Checkout the CMD object in the folder. 
I hope this was the answer you are looking for.

**Comment by tomtefuchs** [comment]
yes, tried this. setting all parameters mentioned in the spec. some are missing in the folder anyway. but no telegram was send according to the log. there is no explicit send button somewhere?

**Comment by Jey-Cee** [comment]
The missing ones are set by the adapter. 
No there is no extra send button.
